microprofile/WFLY-10709-MicroProfile-Tracing-Extension.adoc

```diff
@@ -0,0 +1,79 @@
+= [WFLY-10709] Support Eclipse MicroProfile OpenTracing
+:author:            Juraci Paixão Kröhling
+:email:             jpkroehling@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,tracing,openshift
+
+== Overview
+
+In order to improve Observability of applications deployed in WildFly it would be beneficial to provide
+transparent tracing of Java EE components via automatic initialization of a tracer (Jaeger) plus a set
+of instrumentation libraries for components commonly used in Java EE applications, such as Servlets, JAX-RS,
+CDI, EJB, ...
+
+This will be implemented by supporting Eclipse MicroProfile OpenTracing 1.1 for applications deployed in WildFly.
+The integration uses the smallrye-opentracing component to provide the MicroProfile OpenTracing implementation.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10709[WFLY-10709]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1027[EAP7-1027]
+* https://issues.jboss.org/browse/CLOUD-2728[CLOUD-2728] - Expose EAP "Tracing" information so that it can be consumed by OpenShift Console
+* https://issues.jboss.org/browse/TRACING-85[TRACING-85] - Create jboss-modules for OpenTracing and Jaeger
+* https://issues.jboss.org/browse/TRACING-86[TRACING-86] - OpenTracing integration for EJB in Wildfly
+* https://issues.jboss.org/browse/TRACING-87[TRACING-87] - OpenTracing integration for JAX-RS in Wildfly
+* https://issues.jboss.org/browse/TRACING-88[TRACING-88] - OpenTracing integration for JMS in Wildfly
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tremes@redhat.com[Tomas Remes]
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile OpenTracing is provided by https://github.com/smallrye/smallrye-opentracing[smallrye-opentracing].
+
+=== Other Interested Projects
+
+* A https://github.com/jpkrohling/java-wildfly-tracing[prototype of a WildFly extension] for OpenTracing was developed while exploring a solution for TRACING-85/.../TRACING-88.
+
+== Requirements
+
+* Support Eclipse MicroProfile OpenTracing 1.1 for applications deployed in WildFly.
+* Support a concrete tracer, such as Jaeger's
+
+== Nice-to-Have Requirements
+
+* Provide an option for operators to specify other OpenTracing-compatible tracer implementations than Jaeger's.
+
```

Comment:
How do the TRACING-86 and TRACING-88 issues relate to this?  The spec at [1] looks focused on JAX-RS, so TRACING-87 is covered by the "Support Eclipse MicroProfile OpenTracing 1.1" requirement. Are TRACING-86/88 Nice-to-have (i.e. we'll probably try but it's not a blocker)? Non-Requirements (i.e. explicitly out of scope for this work, but that doesn't mean it can never happen)?

[1] https://github.com/eclipse/microprofile-opentracing/blob/master/spec/src/main/asciidoc/microprofile-opentracing.asciidoc

Reply:
Would it be OK to leave them out for now and, once we have this feature working properly, we add these extra libraries? Or is it preferable to list it all now?

Reply:
It's fine to leave them out, but you should remove the JIRA links so there's no confusion. 

It's no problem doing > 1 of these over time as new functionality gets added over multiple releases.

Generally "Non-Requirements" is used for things an interested reader might assume were required, so you're clearly saying "No it's not." That's not the case here; I doubt a typical reader would assume integrations not part of the spec.

"Nice-to-have" is ambiguous (will it be in or not?) so it's best avoided except for stuff where there's a real good chance of it getting into this iteration of a feature, but maybe not.

Reply:
Got it. I removed those, as well as the nice-to-have requirement.